src/Functions/PartitionByHyperplanes.cpp

```diff
@@ -0,0 +1,98 @@
+#include <base/types.h>
+#include <Columns/ColumnNullable.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnVector.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <Interpreters/Context.h>
+#include "Columns/ColumnArray.h"
+#include "Columns/ColumnFixedString.h"
+#include "Columns/ColumnsNumber.h"
+#include "Core/TypeId.h"
+#include "DataTypes/IDataType.h"
+#include "config.h"
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+}
+
+namespace
+{
+    class FunctionPartitionByHyperplanes : public IFunction
+    {
+    public:
+        static constexpr auto name = "PartitionByHyperplanes";
```

Comment:
`Partition` term has specific meaning in ClickHouse, so please consider different name

Reply:
#48831

Reply:
How would you prefer to name it then? I don't have string opinion, because I just used the name from the issue